pkgs/by-name/vs/vscode-extensions-update/vscode_extensions_update.py

```diff
@@ -0,0 +1,327 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i python3 -p nix python3 nodePackages.semver vsce nix-update gitMinimal
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+
+class VSCodeExtensionUpdater:
+    """
+    A class to update VSCode extension version in nixpkgs.
+    """
+
+    def __init__(self):
+        if not os.path.exists("nixos/release.nix"):
+            print(
+                "Please run this script in the nixpkgs root directory.", file=sys.stderr
+            )
+            sys.exit(1)
+        self.parser = argparse.ArgumentParser(
+            description="Update VSCode extension version in nixpkgs."
+        )
+        self.parser.add_argument(
+            "attribute",
+            default=os.getenv("UPDATE_NIX_ATTR_PATH"),
+            help="nix attribute path of the extension",
+        )
+        self.parser.add_argument(
+            "--override-filename", help="override-filename for nix-update"
+        )
+        self.parser.add_argument(
+            "--pre-release",
+            action="store_true",
+            help="allow check pre-release versions",
+        )
+        self.parser.add_argument(
+            "--platforms",
+            action="store_true",
+            help="set system according to meta.platforms for nix-update",
+        )
+        self.parser.add_argument(
+            "--commit", action="store_true", help="commit the updated package"
+        )
+        self.args = self.parser.parse_args()
+        self.attribute_path = self.args.attribute
+        self.target_vscode_version = self._get_nix_vscode_version()
+        self.current_version = (
+            os.getenv("UPDATE_NIX_OLD_VERSION")
```

Comment:
Should we move this part of the logic in the method  self._get_nix_vscode_extension_version() instead ?